From 71f0fb2bd0b559cee1a4aee330d3e21b23024c3b Mon Sep 17 00:00:00 2001
From: Ron Eldor <Ron.Eldor@arm.com>
Date: Thu, 6 Dec 2018 17:31:55 +0200
Subject: [PATCH 29/43] Change byte copy to memcpy

Change setting the mki value byte after byte with `memcpy()`.
---
 library/ssl_cli.c | 7 ++-----
 library/ssl_srv.c | 7 ++-----
 2 files changed, 4 insertions(+), 10 deletions(-)

diff --git a/library/ssl_cli.c b/library/ssl_cli.c
index 76be7436b..cc9ad7c41 100644
--- a/library/ssl_cli.c
+++ b/library/ssl_cli.c
@@ -677,7 +677,7 @@ static void ssl_write_use_srtp_ext( mbedtls_ssl_context *ssl,
 {
     unsigned char *p = buf;
     size_t protection_profiles_index = 0;
-    size_t mki_len = 0, i;
+    size_t mki_len = 0;
     size_t ext_len = 0;
     uint16_t profile_value = 0;
 
@@ -751,10 +751,7 @@ static void ssl_write_use_srtp_ext( mbedtls_ssl_context *ssl,
 
     if( mki_len != 0 )
     {
-        for( i=0; i < mki_len; i++ )
-        {
-            *p++ = ssl->dtls_srtp_info.mki_value[i];
-        }
+        memcpy( p, ssl->dtls_srtp_info.mki_value, mki_len );
         MBEDTLS_SSL_DEBUG_BUF( 3, "sending mki",  ssl->dtls_srtp_info.mki_value,
                                ssl->dtls_srtp_info.mki_len );
     }
diff --git a/library/ssl_srv.c b/library/ssl_srv.c
index b44c720ce..d704fe2ff 100644
--- a/library/ssl_srv.c
+++ b/library/ssl_srv.c
@@ -2439,7 +2439,7 @@ static void ssl_write_use_srtp_ext( mbedtls_ssl_context *ssl,
                                     unsigned char *buf,
                                     size_t *olen )
 {
-    size_t mki_len = 0, ext_len = 0, i;
+    size_t mki_len = 0, ext_len = 0;
     uint16_t profile_value = 0;
 
     if( ssl->dtls_srtp_info.chosen_dtls_srtp_profile == MBEDTLS_SRTP_UNSET_PROFILE )
@@ -2483,10 +2483,7 @@ static void ssl_write_use_srtp_ext( mbedtls_ssl_context *ssl,
     }
 
     buf[8] = mki_len & 0xFF;
-    for( i=0; i < mki_len; i++ )
-    {
-        buf[9 + i] = ssl->dtls_srtp_info.mki_value[i];
-    }
+    memcpy( &buf[9], ssl->dtls_srtp_info.mki_value, mki_len );
 
     *olen = 9 + mki_len;
 }
-- 
2.17.1

